src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -659,23 +667,42 @@ namespace Opm {
 
             for( const auto& well_name : well_names) {
                 auto& well = this->m_wells.at(well_name);
-                bool addGrupProductionControl = well.isAvailableForGroupControl(currentStep);
-                bool switching_from_injector = !well.isProducer(currentStep);
-                const WellProductionProperties& prev_properties = well.getProductionProperties(currentStep);
-                WellProductionProperties properties(record, isPredictionMode, prev_properties, m_controlModeWHISTCTL, switching_from_injector, addGrupProductionControl);
 
                 updateWellStatus( well , currentStep , status );
-                if (well.setProductionProperties(currentStep, properties)) {
-                    m_events.addEvent( ScheduleEvents::PRODUCTION_UPDATE , currentStep);
-                    this->addWellEvent( well.name(), ScheduleEvents::PRODUCTION_UPDATE, currentStep);
-                }
-                if ( !well.getAllowCrossFlow() && !isPredictionMode && (properties.OilRate + properties.WaterRate + properties.GasRate) == 0 ) {
+                if (isPredictionMode) {
+                    bool addGrupProductionControl = well.isAvailableForGroupControl(currentStep);
+                    WellProductionProperties properties;
 
-                    std::string msg =
+                    if (addGrupProductionControl)
+                        properties.addProductionControl(WellProducer::GRUP);
+
+                    properties.handleWCONPROD(record);
+
+                    if (well.setProductionProperties(currentStep, properties)) {
+                        m_events.addEvent( ScheduleEvents::PRODUCTION_UPDATE , currentStep);
+                        this->addWellEvent( well.name(), ScheduleEvents::PRODUCTION_UPDATE, currentStep);
+                    }
+                } else {
+                    bool switching_from_injector = !well.isProducer(currentStep);
+                    WellProductionProperties properties(well.getProductionProperties(currentStep));
```

Comment:
It turns out that this copying was problematic, because existing control modes were not cleared from the control set. Primitive fix proposed in #746.